packages/golden_toolkit/lib/src/device_builder.dart

```diff
@@ -176,10 +176,19 @@ class _DeviceScenarioWidget extends StatelessWidget {
       device.size.width + (_horizontalScenarioPadding * 2) + (_borderWidth * 2),
       device.size.height + 24 + (_borderWidth * 2));
 
-  Widget get _sizedWidget => Container(
-        width: device.size.width,
-        height: device.size.height,
-        child: widget,
+  Widget get _sizedWidget => MediaQuery(
+        data: MediaQueryData(
```

Comment:
@lunaticcoding sorry about the delay. Can this be modified so that it tries to find a MediaQuery above it and merge them together rather than completely overriding it? My thought here is that a user might be using the `wrap` argument to wrap a widget with `MaterialApp` and I could see this breaking a behavior in their tests.

Reply:
Thanks for the answer :D There is two issues here I think :S 

1) Why would I want to set the mediaquery to anything different than the device size during device golden tests?

2) How would you merge them? If a user wants to set a value that happens to be the default I have no way of knowing that so I would just override it. I could simply not add the media query if the user specifies one but tbh I think that would only occur if the user needs the mediaqueries to be present. (and they would be the same for different devices then :s).

Could you clarify the scenario and the desired solution a bit please? :)

Reply:
In a typical Flutter app the root media query comes from WidgetsApp. It's construction looks like this:
```
MediaQueryData.fromWindow(ui.Window window)
    : size = window.physicalSize / window.devicePixelRatio,
      devicePixelRatio = window.devicePixelRatio,
      textScaleFactor = window.textScaleFactor,
      platformBrightness = window.platformBrightness,
      padding = EdgeInsets.fromWindowPadding(window.padding, window.devicePixelRatio),
      viewPadding = EdgeInsets.fromWindowPadding(window.viewPadding, window.devicePixelRatio),
      viewInsets = EdgeInsets.fromWindowPadding(window.viewInsets, window.devicePixelRatio),
      systemGestureInsets = EdgeInsets.fromWindowPadding(window.systemGestureInsets, window.devicePixelRatio),
      accessibleNavigation = window.accessibilityFeatures.accessibleNavigation,
      invertColors = window.accessibilityFeatures.invertColors,
      disableAnimations = window.accessibilityFeatures.disableAnimations,
      boldText = window.accessibilityFeatures.boldText,
      highContrast = window.accessibilityFeatures.highContrast,
      alwaysUse24HourFormat = window.alwaysUse24HourFormat,
      navigationMode = NavigationMode.traditional;
```

I think in order to make the golden_toolkit work holistically we should be leveraging our `Device` concept to apply to the window. I am **not** asking you to make that change but I think we should consider it while making additions to the golden_toolkit.  As you can see MediaQueryData has more properties than our `Device`. If a user has a test that depends on one of these properties your change will break there tests and they will not have a path to fix it. You can merge the MediaQueryData like this:
```
final mediaQueryData = MediaQuery.of(context, nullOk: true) ?? MediaQueryData();
final mergedMediaQueryData = mediaQuery.copyWith(
          size: device.size,
          padding: device.safeArea,
          platformBrightness: device.brightness,
          devicePixelRatio: device.devicePixelRatio,
          textScaleFactor: device.textScale,
);
```

Reply:
Ooooooooh I see 😅 my bad I’lol also add a test for this :)

Reply:
@fatherOfLegends done :)

Reply:
@fatherOfLegends any update?

Reply:
@fatherOfLegends @coreysprague ? :S

Reply:
I'll take a look. Sorry about that. I had memorized that @fatherOfLegends had taken care of this 😄